src/hooks/requests/useDepartments.ts

```diff
@@ -22,3 +24,73 @@ export const useDepartmentsFetch = () => {
     departmentFilters,
   }
 }
+
+const requestsPromiseThatThrowsAnErrorWhenSomeRequestsFailed = (
+  payload: EditDataType[]
+) =>
+  new Promise(async (resolve, reject) => {
+    const results: PromiseErrorType[] = await Promise.allSettled(
+      map(payload, ({ state, id, name }) => {
+        if (state === 'NEW') {
+          return apiClient.post(endpoints.DEPARTMENTS, { name })
+        }
+        if (state === 'UPDATED') {
+          return apiClient.put(`${endpoints.DEPARTMENTS}/${id}`, { name })
+        }
+        if (state === 'DELETED') {
+          return apiClient.delete(`${endpoints.DEPARTMENTS}/${id}`)
+        }
+      })
+    )
+
+    const fulfilled = compact(
+      map(results, ({ status, value }, key) => {
+        if (status === 'fulfilled') {
+          if (value) {
+            return {
+              key,
+              value,
+            }
+          }
+        }
+      })
+    )
+
+    const errors = compact(
+      map(results, ({ status, reason, value }, key) => {
+        if (status === 'rejected') {
+          const { message, errors: err } = reason || {}
+          const name = `department.${key}`
+          const error = {
+            errors: { [name]: err?.name || [] },
+            message: message || '',
+            name: key,
+          }
+          return error
+        }
+      })
+    )
+
+    if (isEmpty(errors)) {
+      resolve(results)
+    } else {
+      reject([...errors, { values: fulfilled }])
+    }
+  })
+
+export const useParallelMutationDepartment = () => {
+  const queryClient = useQueryClient()
+  const { mutateAsync: parallelUpdating, isLoading } = useMutation({
+    mutationKey: ['departments'],
+    mutationFn: async (payload: EditDataType[]) =>
+      requestsPromiseThatThrowsAnErrorWhenSomeRequestsFailed(payload),
```

Comment:
**[non-blocker]** (Would be a blocker actually)

Basically the ordering of these requests should be revised. At the moment If I try to add a tag with name "Veel 2", but I also remove the previous "Veel 2" department from the list, the deletion will go through, but adding won't since the adding part happens before deletion in some cases, which means that at that point BE thinks that we already have the "Veel 2" tag.

But let's leave it for now, there is no point in putting more time under this part, since we will change the endpoint we use, once we have the bulk edit endpoint anyway.